Cesium.IntegrationTests/type_casts.c

```diff
@@ -0,0 +1,26 @@
+typedef unsigned int size_t;
+
+int foo(int a, int b)
+{
+    return a + b;
+}
+
+int main(void)
+{
+    int a = (int) 1.0;
+
+    size_t b = foo(1, 2);
+    size_t c = (foo)(1, 2);
+
+    if (b + c != 6) return -1;
+
+    size_t d = (size_t) a;
+    size_t e = (size_t) (a + b);
+    if (d + e != 5) return -2;
+
+    int f = (int) a;
+    int g = (int) (a + b);
+    if (f + g != 5) return -3;
+
+    return 42;
```

Comment:
Can you add the `(struct foo)` and `(void*)x` tests and probably `void (*name_func)(int x)`. If they would be broken due to parsing, you can comment out them for now. I just thinking if `TryGetType` would be enough, or we may need something like https://github.com/ForNeVeR/Cesium/pull/382

Reply:
Done

Reply:
Using #382 is fine - I can rebase on that when it gets merged